lib/bolt/cli.rb

```diff
@@ -379,710 +341,601 @@ def validate(options)
+          else
+            outputter.print_tasks(**info)
+          end
+          SUCCESS
         end
-
-        executor.shutdown
-        rerun.update(results)
-
-        outputter.print_summary(results, elapsed_time)
-        code = results.ok ? 0 : 2
       end
-      code
-    rescue Bolt::Error => e
-      outputter.fatal_error(e)
-      raise e
-    ensure
-      # restore original signal handler
-      Signal.trap :INT, handler if handler
-      analytics&.finish
     end
 
-    def show_task(task_name)
-      outputter.print_task_info(pal.get_task(task_name))
-    end
-
-    # Filters a list of content by matching substring.
+    # Process the target list by turning a PuppetDB query or rerun mode into a
+    # list of target names.
     #
-    private def filter_content(content, filter)
-      return content unless content && filter
-      content.select { |name,| name.include?(filter) }
-    end
-
-    def list_tasks
-      tasks = filter_content(pal.list_tasks_with_cache(filter_content: true), options[:filter])
-      outputter.print_tasks(tasks, pal.user_modulepath)
-    end
-
-    def show_plan(plan_name)
-      outputter.print_plan_info(pal.get_plan_info(plan_name))
-    end
-
-    def list_plans
-      plans = filter_content(pal.list_plans_with_cache(filter_content: true), options[:filter])
-      outputter.print_plans(plans, pal.user_modulepath)
-    end
-
-    def list_targets
-      if options.keys.any? { |key| TARGETING_OPTIONS.include?(key) }
-        target_flag = true
-      else
-        options[:targets] = 'all'
-      end
-
-      outputter.print_targets(
-        group_targets_by_source,
-        inventory.source,
-        config.default_inventoryfile,
-        target_flag
-      )
-    end
-
-    def show_targets
-      if options.keys.any? { |key| TARGETING_OPTIONS.include?(key) }
-        target_flag = true
-      else
-        options[:targets] = 'all'
-      end
-
-      outputter.print_target_info(
-        group_targets_by_source,
-        inventory.source,
-        config.default_inventoryfile,
-        target_flag
-      )
-    end
-
-    # Returns a hash of targets sorted by those that are found in the
-    # inventory and those that are provided on the command line.
+    # @param pdb_client [Bolt::PuppetDB::Client] The PuppetDB client.
+    # @param rerun [Bolt::Rerun] The Rerun instance.
+    # @param options [Hash] The CLI options.
     #
-    private def group_targets_by_source
-      # Retrieve the known group and target names. This needs to be done before
-      # updating targets, as that will add adhoc targets to the inventory.
-      known_names = inventory.target_names
-
-      update_targets(options)
-
-      inventory_targets, adhoc_targets = options[:targets].partition do |target|
-        known_names.include?(target.name)
-      end
-
-      { inventory: inventory_targets, adhoc: adhoc_targets }
-    end
+    private def process_target_list(pdb_client, rerun, options)
```

Comment:
Ah I see - can we check for multiple targeting options here? Targeting options can be optional, but anything that does accept targeting options should only ever have one specified.